python/lsst/meas/algorithms/readTextCatalogTask.py

```diff
@@ -54,6 +54,20 @@ class ReadTextCatalogConfig(pexConfig.Config):
         doc=("Format of files to read, from the astropy.table I/O list here:"
              "http://docs.astropy.org/en/stable/io/unified.html#built-in-table-readers-writers")
     )
+    fill_values = pexConfig.ListField(
+        dtype=str,
+        default=None,
+        optional=True,
+        doc=("A list giving [<match_string>, <fill_value>], which is used to mask"
+             " the given values in the input file. '0' is suggested for the fill value in order to prevent"
+             "changing the column datatype. The default behavior is to fill empty data with zeros. See "
+             "https://docs.astropy.org/en/stable/io/ascii/read.html#bad-or-missing-values for more details.")
+    )
+    replace_missing_floats = pexConfig.Field(
+        dtype=float,
+        optional=True,
+        doc="If True, replace missing data in float columns with given value instead of zero."
```

Comment:
The docstring doesn't match the behavior below: below, you always fill missing floats with NaN, which I'd argue is the correct behavior (instead of "with the given value"). Null float fields should become NaN for the purposes of this code. `replace_missing_floats` should then just be a boolean field.

Reply:
Yeah, I started with this being a boolean, but then I thought I should just leave it up to the user, so I changed it to a float field (and then forgot to actually change the code). NaN is of course the most likely value wanted here, but I don't think there's any harm in allowing flexibility in choosing the replacement value here.

Reply:
I think this should always fill with NaN. Making this configurable is harder for the user (e.g. they have to import numpy and set this to np.nan), and NaN is 100% the correct floating point value for "there is no value".